--- a/usr/share/lipstick-jolla-home-qt5/compositor.qml	2015-03-04 14:13:18.979270182 +0200
+++ b/usr/share/lipstick-jolla-home-qt5/compositor.qml	2015-03-05 08:56:59.010067203 +0200
@@ -94,6 +94,11 @@
     // The topmost Alarm window (e.g. incoming voice call, clock alarm)
     property Item topmostAlarmWindow
 
+    property bool topmostIsAndroidApp: (
+        topmostApplicationWindow &&
+        (topmostApplicationWindow.window.surface.className.substr(-8) != '.desktop')
+    )
+
     property int eventsWindowOrientation: Qt.PortraitOrientation
     property bool eventsViewNeedsConnectivity
     property int inProcessOrientation: (eventPeek || eventsActive) ? eventsWindowOrientation : Qt.PortraitOrientation
@@ -107,26 +112,34 @@
     onTopmostDialogWindowChanged: updateWindows()
     onTopmostAlarmWindowChanged: updateWindows()
     onTopmostWindowOrientationChanged: {
+        PeekFilter.topEnabled = root.topPeekAllowed
+        PeekFilter.leftEnabled = root.leftPeekAllowed
+        PeekFilter.rightEnabled = root.rightPeekAllowed
+        PeekFilter.bottomEnabled = root.bottomPeekAllowed
         switch (topmostWindowOrientation) {
         case Qt.LandscapeOrientation:
+            PeekFilter.topEnabled = !topmostIsAndroidApp
             leftArea = PeekFilter.Top
             rightArea = PeekFilter.Bottom
             topArea = PeekFilter.Right
             bottomArea = PeekFilter.Left
             break;
         case Qt.InvertedPortraitOrientation:
+            PeekFilter.rightEnabled = !topmostIsAndroidApp
             leftArea = PeekFilter.Right
             rightArea = PeekFilter.Left
             topArea = PeekFilter.Bottom
             bottomArea = PeekFilter.Top
             break;
         case Qt.InvertedLandscapeOrientation:
+            PeekFilter.bottomEnabled = !topmostIsAndroidApp
             leftArea = PeekFilter.Bottom
             rightArea = PeekFilter.Top
             topArea = PeekFilter.Left
             bottomArea = PeekFilter.Right
             break;
         default:
+            PeekFilter.leftEnabled = !topmostIsAndroidApp
             leftArea = PeekFilter.Left
             rightArea = PeekFilter.Right
             topArea = PeekFilter.Top
