--- a/usr/share/lipstick-jolla-home-qt5/compositor.qml	2015-05-06 19:51:02.536330083 +0300
+++ b/usr/share/lipstick-jolla-home-qt5/compositor.qml	2015-05-07 15:12:30.694360788 +0300
@@ -101,7 +101,7 @@
 
     PeekFilter {
         id: globalPeekFilterItem
-        leftEnabled: true
+        leftEnabled: appLayer.active ? !appLayer.isAndroidApp : true
         rightEnabled: true
         topEnabled: true
         bottomEnabled: true
@@ -230,6 +230,11 @@
             if (appLayer.window && appLayer.window != w) appLayer.hideWindow()
             appLayer.window = w
             appLayer.window.visible = true
+            appLayer.isAndroidApp = (
+                w.window &&
+                w.window.surface &&
+                (w.window.surface.className.substr(-8) != ".desktop")
+            )
             w.window.takeFocus()
         }


--- a/usr/share/lipstick-jolla-home-qt5/layers/AppLayer.qml	2015-05-06 20:04:10.010041389 +0300
+++ b/usr/share/lipstick-jolla-home-qt5/layers/AppLayer.qml	2015-05-07 12:04:52.761906618 +0300
@@ -12,6 +12,8 @@
     readonly property bool closePeek: peekFilter.topActive
     readonly property bool switching: switchApplicationAnimation.running
 
+    property bool isAndroidApp: false
+
     function switchToApp(w) {
         switchApplicationAnimation.fromWindow = window
         switchApplicationAnimation.toWindow = w
@@ -97,7 +99,7 @@
     PeekFilter {
         id: appPeekFilter
         enabled: appLayer.active
-        leftEnabled: true
+        leftEnabled: !isAndroidApp
         rightEnabled: true
         topEnabled: true
         onGestureTriggered: {
